components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/SignKeyDataHolder.java

```diff
@@ -144,8 +150,14 @@ public SignKeyDataHolder(String username) throws IdentityException {
                 }
             }
 
+        } catch (IdentityException e) {
+            throw new IdentityException("Unable to access realm service", e);
```

Comment:
Need to put more context information on the message. e.g. UserName, key alias, tenantDomain, etc.
So that anyone looking through the logs will be able to guess what is the cause for the error.